rfcs/20220221-3d_model_support.md

```diff
@@ -0,0 +1,181 @@
+
+## Prior Art
+
+- We already show 3D models as [part of images](https://search.openverse.engineering/search/image?q=dogs&source=sketchfab,thingiverse), which is great! Fewer scripts to write on the catalog side.
+- We also already show Sketchfab's 3D model viewer for their single results: https://search-staging.openverse.engineering/image/00a6a7a4-6605-4cba-bacf-a08b0da546b9
+
+## Some code considerations
+
+- When a unique identifier is necessary, we should use the string constant `const MODEL_3D = '3d model'` to refer to 3D models. Compare this to `AUDIO` for audio files, for example.
+- Similar to the catalog/API, 3D model support should be enabled via a feature flag. This work should be able to be committed directly to `main` without a base feature branch needed. Meta search support for 3D models does not need to be feature flagged.
+- e2e tests should be added for any new routes; and updated for existing routes that are modified. PRs will be blocked in the abcense of these tests.
+- New content types should always launch in `beta` until we reach a certain number of results and/or are confident in our UI components (this metric(s) need to be determined).
+- Many of the todos identified here can be broken down into even smaller pieces in the future, particularly around the user interface.
+
+## Technical implementation
+
+### 1. Add 3D Models to Meta Search
+
+#### Prerequisites
+
+- Need a list of 3D model providers to use in meta search. This could be a mix of sources we intend to use in the Catalog but also some that don't have APIs and will stay in meta search after launching 3D model support.
+
+#### Todos
+
+- [ ] Add 3D models to the content switcher components:
+  - [ ] https://github.com/WordPress/openverse-frontend/blob/bbccc36ed49186b369cc4a69693fbc6d1411f29a/src/components/VContentSwitcher/VContentSwitcherButton.vue
+  - [ ] https://github.com/WordPress/openverse-frontend/blob/7d74f033ddc4d35889d7c43756c36150beee4192/src/components/VContentSwitcher/VContentTypes.vue
+- [ ] Create a `model/index.vue` endpoint for 3D models in the [pages directory](https://github.com/WordPress/openverse-frontend/blob/037f2efb768fdbdcff4b875e7f7c271ff50df59f/src/pages)
+- [ ] Add 3D model sources to the [legacy sources lookup](https://github.com/WordPress/openverse-frontend/blob/268532a05004b72fb91b77a875f3a3c1d7fb0599/src/utils/get-legacy-source-url.js#L1)
+- [ ] Add 3D models to the [media constants](https://github.com/WordPress/openverse-frontend/blob/4aeb9fd0d9f893f2df1752d6bb1e52351b8c3d35/src/constants/media.js#L13) file, making sure it is set to the `ADDITIONAL` status.
+
+### 2. Setup the data for 3D models
+
+- [ ] Create a feature flag for 3D model support. All 3D model features will be behind this.
```

Comment:
Can we specify what "Create a feature flag" means exactly _vis_ https://github.com/WordPress/openverse/pull/171/files#r811500414. I feel like regardless of the _how_ behind the feature flag flipping mechanism it should get put inside a composable, right? Probably a `readonly`-after-initialization Pinia store?

Reply:
Added a note that this depends on the feature flag RFC.